src/raftstore/store/worker/apply.rs

```diff
@@ -2144,6 +2323,58 @@ impl Runner {
         }
     }
 
+    fn handle_resume(&mut self, r: Resume) {
+        let mut delegate = match self.delegates.get_mut(&r.region_id) {
+            Some(e) => e.take().unwrap(),
+            None => panic!("[region {}] is missing", r.region_id),
+        };
+        assert!(delegate.is_pause());
+
+        let mut core = ApplyContextCore::new(
+            self.host.as_ref(),
+            self.importer.as_ref(),
+            Some(self.sub_apply_worker.scheduler()),
+        ).use_delete_range(self.use_delete_range)
+            .enable_sync_log(self.sync_log);
+
+        // Resume the region.
+        let mut pause_point = delegate.take_pause_point();
+        core.prepare_for(&delegate);
+        core.cbs.last_mut().unwrap().push(
+            pause_point.cur_cb.take(),
+            pause_point.cur_resp.take().unwrap(),
+        );
+        core.finish_for(&mut delegate, pause_point.results);
+
+        // Handle remain entries.
+        let apply = Apply::new(r.region_id, delegate.term, pause_point.remain_entries);
+        delegate.metrics = ApplyMetrics::default();
+        delegate.term = apply.term;
+
+        {
+            let mut ctx = ApplyContext::new(&mut core, &mut self.delegates);
+            delegate.handle_raft_committed_entries(&mut ctx, apply.entries);
```

Comment:
Will there be a recursive problem? maybe when we resume, we find another `pause`?

Reply:
Recursion is ok here?